Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,90 @@
+require "set"
+
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.10.1.tar.gz"
+  sha256 "e571382dbccca0c37f38d097a12bf54874190a312e55ce22534cc6b1ddb8ffaf"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  WITHOUT_OPTIONS = {
+    "compute" => ["ec2", "elasticbeanstalk", "lambda"],
+    "storage" => ["s3", "cloudfront", "elasticfilesystem", "glacier", "importexport", "storagegateway"],
+    "database" => ["rds", "dynamodb", "elasticache", "redshift", "dms", "sdb"],
+    "networking" => ["directconnect", "route53", "route53domains"],
+    "devtools" => ["codecommit", "codedeploy", "codepipeline"],
+    "management" => ["cloudformation", "cloudtrail", "config", "opsworks", "ecr", "ecs", "autoscaling", "support", "logs", "monitoring", "ssm"],
+    "security" => ["identity-management", "access-management", "inspector", "waf", "iam", "cloudhsm", "acm", "ds", "kms", "sts"],
+    "analytics" => ["datapipeline", "kinesis", "firehose", "marketplacecommerceanalytics"],
+    "mobile" => ["cognito-identity", "cognito-sync", "devicefarm", "mobileanalytics", "sns"],
+    "appservices" => ["apigateway", "cloudsearch", "cloudsearchdomain", "elastictranscoder", "sqs", "swf", "email"],
+    "other" => ["elasticloadbalancing", "elasticmapreduce", "es", "events", "machinelearning", "queues", "transfer", "iot", "gamelift", "workspaces", "meteringmarketplace"],
+  }.freeze
+
+  WITHOUT_OPTIONS.each { |category, clients| option "without-#{category}", "Exclude clients: #{clients.join(", ")}" }
```

Comment:
Hello folks! Just passing by as I'd also like to build some tools with the AWS SDK. I love using homebrew to setup development environments-- and having these somewhat-expressive options will be perfect.

Don't options exist for the benefit of the from-source builder? Exposing these options seems very reasonable given the myriad of ways I see people using brew.

I guess the question for @zwass is, can you imagine a use for each of the disablements listed? I don't know if it's totally appropriate for him to answer. From what I can tell he's not an AWS SDK developer, just a SDK user.